From 43acf278c051fe746893df04ce4ad7a971ee5e72 Mon Sep 17 00:00:00 2001
From: Brandon Huang <yanghuan@codeaurora.org>
Date: Tue, 20 Apr 2021 14:46:01 +0800
Subject: [PATCH] ath10k: fix tx data rate in legacy mode

In legacy mode, ath10k adopts a u8 variable to calculate legacy
tx data rate, but this variable is overflow when data rate is
high. Fix this issue by convert data type from u8 to u16.

Change-Id: I7b4df0a13e3c7b96988603a8815209a53014d102
Signed-off-by: Brandon Huang <yanghuan@codeaurora.org>
---
 drivers/net/wireless/ath/ath10k/htt_rx.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/net/wireless/ath/ath10k/htt_rx.c b/drivers/net/wireless/ath/ath10k/htt_rx.c
index de141c9..f0b9802 100644
--- a/drivers/net/wireless/ath/ath10k/htt_rx.c
+++ b/drivers/net/wireless/ath/ath10k/htt_rx.c
@@ -2222,7 +2222,8 @@ ath10k_update_per_peer_tx_stats(struct ath10k *ar,
 				struct ath10k_per_peer_tx_stats *peer_stats)
 {
 	struct ath10k_sta *arsta = (struct ath10k_sta *)sta->drv_priv;
-	u8 rate = 0, sgi;
+	u8 sgi;
+	u16 rate = 0;
 	struct rate_info txrate = {};
 
 	lockdep_assert_held(&ar->data_lock);
-- 
2.7.4

